src/link/batch/batching.ts

```diff
@@ -9,23 +9,31 @@ export type BatchHandler = (
 export interface BatchableRequest {
   operation: Operation;
   forward?: NextLink;
-
-  // promise is created when the query fetch request is
-  // added to the queue and is resolved once the result is back
-  // from the server.
   observable?: Observable<FetchResult>;
   next?: Array<(result: FetchResult) => void>;
   error?: Array<(error: Error) => void>;
   complete?: Array<() => void>;
 }
 
+interface QueuedRequest extends BatchableRequest {
+  next: Array<(result: FetchResult) => void>;
+  error: Array<(error: Error) => void>;
+  complete: Array<() => void>;
+  subscribers: Set<object>;
+}
```

Comment:
What do you think about this type
```suggestion
type QueuedRequest =
  BatchableRequest &
  Required<Pick<BatchableRequest, 'next' | 'error' | 'complete'>> &
  { subscribers?: Set<object> }
```

Reply:
Following https://github.com/apollographql/apollo-client/pull/9248/commits/29fb39ed033614aa2d441034acda32c51545171b, this suggestion is not applicable anymore